Products/CMFPlone/browser/recyclebin.py

```diff
@@ -0,0 +1,622 @@
+            except Exception as e:
+                logger.error(f"Error restoring child item: {e}")
+                message = "Failed to restore child item."
+                IStatusMessage(self.request).addStatusMessage(message, type="error")
+
+    def get_item(self):
+        """Get the specific recycled item"""
+        logger.info(f"RecycleBinItemView.get_item called for ID: {self.item_id}")
+        if not self.item_id:
+            logger.warning("get_item called with no item_id")
+            return None
+
+        recycle_bin = getUtility(IRecycleBin)
+        item = recycle_bin.get_item(self.item_id)
+        if item is None:
+            logger.warning(f"No item found in recycle bin with ID: {self.item_id}")
+        else:
+            logger.info(
+                f"Found item: {item.get('title', 'Unknown')} of type {item.get('type', 'Unknown')}"
+            )
+        return item
+
+    def get_children(self):
+        """Get the children of this item if it's a folder or collection"""
+        item = self.get_item()
+        if item and "children" in item:
+            children = []
+            for child_id, child_data in item["children"].items():
+                # Don't include the actual object in the listing
+                child_info = child_data.copy()
+                if "object" in child_info:
+                    del child_info["object"]
+                children.append(child_info)
+            return children
+        return []
+
+    def get_comment_children(self):
+        """Get comments from a CommentTree item"""
+        item = self.get_item()
+        if item and item.get("type") == "CommentTree":
+            comment_tree = item.get("object", {})
+            comments = comment_tree.get("comments", [])
+
+            # Process comments to build a list for display
+            comment_list = []
+            for comment_obj, comment_path in comments:
+                # Extract comment data
+                comment_data = {
+                    "id": getattr(comment_obj, "comment_id", ""),
+                    "text": getattr(comment_obj, "text", ""),
+                    "author": getattr(comment_obj, "author_name", None)
+                    or getattr(comment_obj, "author_username", "Anonymous"),
+                    "in_reply_to": getattr(comment_obj, "in_reply_to", None),
+                    "path": comment_path,
+                    "creation_date": getattr(comment_obj, "creation_date", None),
+                    "modification_date": getattr(
+                        comment_obj, "modification_date", None
+                    ),
+                    "title": f"Comment by {getattr(comment_obj, 'author_name', None) or getattr(comment_obj, 'author_username', 'Anonymous')}",
+                    "size": len(getattr(comment_obj, "text", "")),
+                }
+                comment_list.append(comment_data)
+
+            return comment_list
+        return []
+
+    def format_date(self, date):
```

Comment:
Use directly in template. see above.

Reply:
done https://github.com/plone/Products.CMFPlone/pull/4168/commits/55e2b9a5f5f5602fccaee2833e29c04bdd86c94a